### PR TITLE
[APK] Refactor publisher service impls

### PR DESCRIPTION
**Pull request by RakhithaRR**
This PR refactors the code in following service impls.

- ApiCategoriesApiServiceImpl
- ApiProductsApiServiceImpl
- ApisApiServiceImpl
- EndpointCertificatesApiServiceImpl
- ExternalStoresApiServiceImpl
- KeyManagersApiServiceImpl
- MeApiServiceImpl
- OperationPoliciesApiServiceImpl

### CODE REVIEW

File: components/apimgt/org.wso2.carbon.apimgt.api/src/main/java/org/wso2/carbon/apimgt/api/dto/APIEndpointValidationDTO.java [hunk 1]
```diff
@@ -0,0 +1,35 @@
+package org.wso2.carbon.apimgt.api.dto;
```
Comment (tharindu1st) [inline components/apimgt/org.wso2.carbon.apimgt.api/src/main/java/org/wso2/carbon/apimgt/api/dto/APIEndpointValidationDTO.java:1]: license header

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/restapi/CommonUtils.java [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+import org.wso2.carbon.apimgt.api.APIProvider;
+import org.wso2.carbon.apimgt.api.ExceptionCodes;
+import org.wso2.carbon.apimgt.api.model.API;
+import org.wso2.carbon.apimgt.api.model.Scope;
+import org.wso2.carbon.apimgt.impl.utils.APIUtil;
+
+import java.util.HashSet;
+import java.util.Set;
+
+public class CommonUtils {
```
Comment (tharindu1st) [inline components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/restapi/CommonUtils.java:14]: class comment

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/restapi/CommonUtils.java [hunk 3]
```diff
@@ -0,0 +1,80 @@
+package org.wso2.carbon.apimgt.impl.restapi;
+
```
Comment (tharindu1st) [inline components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/restapi/CommonUtils.java:2]: license

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/restapi/CommonUtils.java [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+                    .append(endpointType)
+                    .append("\",")
+                    .append("\"production_endpoints\": {\"url\": \"")
+                    .append(serviceUrl)
+                    .append("\"}}");
+        } // TODO Need to check on the endpoint security
+        return sb.toString();
+    }
+
+    public static void validateScopes(API api, APIProvider apiProvider, String username) throws APIManagementException {
```
Comment (tharindu1st) [inline components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/restapi/CommonUtils.java:34]: method commend

**Review by tharindu1st — CHANGES_REQUESTED** [review]

File: components/apimgt/org.wso2.carbon.apimgt.api/src/main/java/org/wso2/carbon/apimgt/api/dto/APIEndpointValidationDTO.java (hunk 1 above)
Comment (RakhithaRR) [reply components/apimgt/org.wso2.carbon.apimgt.api/src/main/java/org/wso2/carbon/apimgt/api/dto/APIEndpointValidationDTO.java:1]: Updated with https://github.com/wso2/carbon-apimgt/pull/11435/commits/2ca8fb65474e6e00dc257b51c8444c5067239cb6

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/restapi/CommonUtils.java (hunk 2 above)
Comment (RakhithaRR) [reply components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/restapi/CommonUtils.java:36]: Updated with https://github.com/wso2/carbon-apimgt/pull/11435/commits/2ca8fb65474e6e00dc257b51c8444c5067239cb6

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/restapi/CommonUtils.java (hunk 3 above)
Comment (RakhithaRR) [reply components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/restapi/CommonUtils.java:20]: Updated with https://github.com/wso2/carbon-apimgt/pull/11435/commits/2ca8fb65474e6e00dc257b51c8444c5067239cb6

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/restapi/CommonUtils.java (hunk 4 above)
Comment (RakhithaRR) [reply components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/restapi/CommonUtils.java:67]: Updated with https://github.com/wso2/carbon-apimgt/pull/11435/commits/2ca8fb65474e6e00dc257b51c8444c5067239cb6

**Review by tharindu1st — APPROVED** [review]

**Pull request merged**